framework/caching/Cache.php

```diff
@@ -109,6 +109,7 @@ public function buildKey($key)
         if (is_string($key)) {
             $key = ctype_alnum($key) && StringHelper::byteLength($key) <= 32 ? $key : md5($key);
         } else {
+            $key = $this->convertArrayValuesToString((array) $key);
```

Comment:
I would wrap it in `if (is_array())` since $key can be whatever.

Reply:
It is typecast to array first.

Reply:
Yes, I think we should avoid it.

Reply:
@samdark do you want me to check if key is an array?

Reply:
Would something less costy not suffice ? e.g. `$key = md5(var_export($key))`; 
Recursive calculation of a simple cache key seems like quite overhead ... :-)

Reply:
> Would something less costy not suffice ? e.g. `$key = md5(var_export($key))`;
> Recursive calculation of a simple cache key seems like quite overhead ... :-)

`var_export(1) !== var_export('1')`

Plus it is only recursive if your keys are n+1 dimensional array

Reply:
What about `print_r(1, true) === print_r('1', true)` ? Is more [type agnostic](https://3v4l.org/RoG6Y).

Reply:
if print_r works,(let me test that) then I will probably use that. I would like to benchmark it as well to see which is actually more performant

Reply:
so, results are in

`print_r()` won my competition, and wasn't even slower than previous `serialize()`, therefore in my eyes, it is better solution.

My solution was 20% slower

Reply:
It will be definitely longer. As I think it should work well with objects also, can anyone see any potential problems with object keys?

Reply:
no, it will not be longer (storage-wise), because keys are `md5` anyway

Reply:
Oh, right, true. Question still stands.

Reply:
Waw, I am actually surprised that tests passed, I had doubts about objects. 
I found one [use case to verify](https://3v4l.org/FpGbO) - type juggling (`null - false - 0 - "0"`) - but IMHO all 3 false results are justified, or at least same as current implementation.  
Still I would suggest to test also under `strict_types=1` mode if that does not bring up some unexpected issues.